File: JuegoEnJavayRubyNapakalaki/Napakalaki/src/NapakalakiGame/Treasure.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package NapakalakiGame;

/**
 *
 * @author nen155
 */
public class Treasure implements Card {
   private String name;
   private int goldCoins;
   private int minBonus;
   private int maxBonus;
   private TreasureKind type;
  

    public Treasure(String name, int goldCoins, int minBonus, int maxBonus, TreasureKind type) {
        this.name = name;
        this.goldCoins = goldCoins;
        this.minBonus = minBonus;
        this.maxBonus = maxBonus;
        this.type = type;
    }
   
    public int getGoldCoins() {
        return goldCoins;
    }

    public int getMinBonus() {
        return minBonus;
    }

    public int getMaxBonus() {
        return maxBonus;
    }

    public TreasureKind getType() {
        return type;
    }
   
    public String getName() {
        return name;
    }
    public String toString(){
        return "Nombre: "+name+" Monedas: "+goldCoins+" MinBonus: "+ minBonus+ " MaxBonus: "+ maxBonus + " Tipo: "+type;
    }

    @Override
    public int getBasicValue() {
        //throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
        return minBonus;
    }

    @Override
    public int getSpecialValue() {
        //throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
        return maxBonus;
    }
}

File: JuegoEnJavayRubyNapakalaki/Napakalaki_Ruby/lib/treasure.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
require_relative 'card'
module Napakalaki
  class Treasure < Card
    attr_reader :name,:goldCoins,:minBonus,:maxBonus,:type
    def initialize(n,g,min,max,t)
      @name=n
      @goldCoins=g
      @minBonus=min
      @maxBonus=max
      @type=t
    end
    def to_s
      "Nombre: "+@name+" Monedas: "+@goldCoins.to_s+" MinBonus: "+ @minBonus.to_s+ " MaxBonus: "+ @maxBonus.to_s + " Tipo: "+@type
    end
  end
end
